documentation/source/configuration_files.md

```diff
@@ -0,0 +1,248 @@
+Here are a few simple examples:
+```yaml
+a: 1
+b: 2
+c: 3
+a_plus_b: "${add: ${a},${b}}"
+a_plus_b_plus_c: "${add: ${a}, ${b}, ${c}}"
+                 
+my_list: [10, 20, 30, 40, 50]
+third_of_list: "${getitem: ${my_list}, 2}"
+first_of_list: "${first: ${my_list}}"
+last_of_list: "${last: ${my_list}}"
+```
+
+The more advanced resolvers will instantiate objects. In the following example we define a few transforms that 
+will be used to augment a dataset.
+```yaml
+train_dataset_params:
+  transforms:
+    # for more options see common.factories.transforms_factory.py
+    - SegColorJitter:
+        brightness: 0.1
+        contrast: 0.1
+        saturation: 0.1
+
+    - SegRandomFlip:
+        prob: 0.5
+
+    - SegRandomRescale:
+        scales: [ 0.4, 1.6 ]
+```
+Each one of the keys (`SegColorJitter`, `SegRandomFlip`, `SegRandomRescale`) is mapped to a type, and the configuration parameters under that key will be passed
+to the type constructor by name (as key word arguments).
+
+If you want to see where this magic is happening, you can look for the `@resolve_param` decorator in the code 
+
+```python
+class ImageNetDataset(torch_datasets.ImageFolder):
+    
+    @resolve_param("transforms", factory=TransformsFactory())
+    def __init__(self, root: str, transforms: Union[list, dict] = [], *args, **kwargs):
+        ...
+        ...
+```
+
+The `@resolve_param` wraps functions and resolves a string or a dictionary argument (in the example above "transforms") to an object. 
+To do so, it uses a factory object that maps a string or a dictionary to a type. when `__init__(..)` will be called, the function will receive 
+an object, and not a dictionary. The parameters under "transforms" in the YAML will be passed as
+arguments for instantiation the objects. We will  learn how to add a new type of object into these mappings in the next sections. 
+
+## Registering a new object 
+To use a new object from your configuration file, you need to define the mapping of the string to a type.
+This is done using one of the many registration function supported by SG.
+```python
+register_model
+register_detection_module
+register_metric
+register_loss
+register_dataloader
+register_callback
+register_transform
+register_dataset
+```
+
+These decorator functions can be imported and used as follows:
+
+```python
+from super_gradients.common.registry import register_model
+
+@register_model(name="MyNet")
+class MyExampleNet(nn.Module):
+    def __init__(self, num_classes: int):
+        ....
+```
+
+This simple decorator, maps the name "MyNet" to the type `MyExampleNet`. Note that if your constructor
+include required arguments, you will be expected to provide them when using this string
+
+```yaml
+...
+architecture: MyNet
+    num_classes: 8
+...
+
+```
+
+## Recipes library structure
+The `super_gradients/recipes` include the following subdirectories:
```

Comment:
I think should mention the other parameters that are at the top level (num gpu, multi_gpu..)- otherwise users might crash their code.